imagetest/test_suites/oslogin/oslogin_utils.go

```diff
@@ -0,0 +1,199 @@
+}
+
+// Checks if OSLogin is enabled. Returns an error if it is not, or there is trouble
+// reading a file.
+func isOsLoginEnabled(ctx context.Context) error {
+	data, err := os.ReadFile("/etc/nsswitch.conf")
+	if err != nil {
+		return fmt.Errorf("cannot read /etc/nsswitch.conf")
+	}
+	for _, line := range strings.Split(string(data), "\n") {
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		if strings.Contains(line, "passwd:") && !strings.Contains(line, "oslogin") {
+			return fmt.Errorf("OS Login not enabled in /etc/nsswitch.conf")
+		}
+	}
+
+	// Check AuthorizedKeys Command
+	data, err = os.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return fmt.Errorf("cannot read /etc/ssh/sshd_config")
+	}
+	var foundAuthorizedKeys bool
+	for _, line := range strings.Split(string(data), "\n") {
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		if strings.Contains(line, "AuthorizedKeysCommand") && strings.Contains(line, "/usr/bin/google_authorized_keys") {
+			foundAuthorizedKeys = true
+		}
+	}
+
+	if !foundAuthorizedKeys {
+		return fmt.Errorf("AuthorizedKeysCommand not set up for OS Login")
+	}
+
+	if err = testSSHDPamConfig(ctx); err != nil {
+		return fmt.Errorf("error checking pam config: %v", err)
+	}
+	return nil
+}
+
+func testSSHDPamConfig(ctx context.Context) error {
+	twoFactorAuthEnabled, err := isTwoFactorAuthEnabled(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to query two factor authentication metadata entry: %+v", err)
+	}
+
+	if twoFactorAuthEnabled {
+		// Check Pam Modules
+		data, err := os.ReadFile("/etc/pam.d/sshd")
+		if err != nil {
+			return fmt.Errorf("cannot read /etc/pam.d/sshd: %+v", err)
+		}
+
+		if !strings.Contains(string(data), "pam_oslogin_login.so") {
+			return fmt.Errorf("OS Login PAM module missing from pam.d/sshd")
+		}
+	}
+	return nil
+}
+
+func isTwoFactorAuthEnabled(ctx context.Context) (bool, error) {
+	var (
+		instanceFlag, projectFlag bool
+		err                       error
+	)
+
+	elem := []string{"attributes", "enable-oslogin-2fa"}
+
+	instanceFlag, err = getTwoFactorAuth(ctx, "instance", elem...)
+	if err != nil && !errors.Is(err, utils.ErrMDSEntryNotFound) {
+		return false, err
+	}
+	projectFlag, err = getTwoFactorAuth(ctx, "project", elem...)
+	if err != nil && !errors.Is(err, utils.ErrMDSEntryNotFound) {
+		return false, err
+	}
+	return instanceFlag || projectFlag, nil
+}
+
+func getTwoFactorAuth(ctx context.Context, root string, elem ...string) (bool, error) {
+	data, err := utils.GetMetadata(ctx, append([]string{root}, elem...)...)
+	if err != nil {
+		return false, err
+	}
+	flag, err := strconv.ParseBool(data)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse enable-oslogin-2fa metadata entry: %+v", err)
+	}
+	return flag, nil
+}
+
+// Gets the given secret.
+func getSecret(ctx context.Context, client *secretmanager.Client, secretName string) (string, error) {
```

Comment:
I don't believe any other tests are using secrets right now, but I understand what you mean. I will move it to the more general utils file since it's a more general function rather than something that is specifically built for OSLogin testing.